app/labor/rate_limit_checker.rb

```diff
@@ -10,6 +10,8 @@ def limit_by_situation(situation)
                user.comments.where("created_at > ?", 30.seconds.ago).size > 9
              when "published_article_creation"
                user.articles.published.where("created_at > ?", 30.seconds.ago).size > 9
+             when "image_upload"
+               Rails.cache.read("#{user.id}_image_upload") > 10
```

Comment:
You would have to do `to_i` here too since the initial cache read would return nil.
```suggestion
               Rails.cache.read("#{user.id}_image_upload").to_i > 10
```